src/main/kotlin/io/zerobase/smarttracing/resources/OrganizationsResource.kt

```diff
@@ -1,47 +1,150 @@
+    @JsonProperty("isTesting") val isTesting: Boolean,
+    @JsonProperty("contact") val contact: Contact?
+)
+
+data class CreateScannableRequest(
+    @JsonProperty("type") val type: String,
+    @JsonProperty("singleUse") val singleUse: Boolean
+)
 
 @Path("/organizations")
 @Consumes(MediaType.APPLICATION_JSON)
 @Produces(MediaType.APPLICATION_JSON)
-class OrganizationsResource {
+@SuppressFBWarnings("RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE", justification = "false positive")
+class OrganizationsResource(val dao: GraphDao, private val siteTypes: MultiMap<String, String>, private val scanTypes: List<String>) {
 
     @POST
     @Creator
-    fun createOrganization(request: Any): IdWrapper {
-        TODO("not implemented")
+    fun createOrganization(request: CreateOrganizationRequest): IdWrapper? {
+        val name = request.name
+        val phone = request.contact.phone
+        val email = request.contact.email
+        val contactName = request.contact.contactName
+        val address = request.address
+        val hasTestingFacilities = request.hasTestingFacilities ?: false
+        val hasMultipleSites = request.hasMultipleSites ?: true
+
+        val id = dao.createOrganization(
+            name, phone, email,
+            contactName, address,
+            hasTestingFacilities,
+            hasMultipleSites
+        )
+
+        return when (id) {
+            null -> null
+            else -> IdWrapper(id)
+        }
     }
 
     @Path("/{id}/sites")
     @POST
     @Creator
-    fun createSite(@PathParam("id") id: String, request: Any): IdWrapper {
-        TODO("not implemented")
+    fun createSite(@PathParam("id") id: String, request: CreateSiteRequest): IdWrapper? {
+        val name = request.name ?: ""
+        val category = request.category
+        val subcategory = request.subcategory
+        val latitude = request.location.latitude
+        val longitude = request.location.longitude
+        val isTesting = request.isTesting
+        val isCat = siteTypes.containsKey(category)
+        val isSubCat = siteTypes[category]?.contains(subcategory)
+
+        if (!(isCat!!)) {
+            return null
+        }
+
+        if (!(isSubCat!!)) {
+            return null
+        }
+
+        val id = if (request.contact == null) {
+            dao.createSite(
+                id, name, category, subcategory, latitude,
+                longitude, isTesting, null, null, null
+            )
+        } else {
+            dao.createSite(
+                id, name, category, subcategory, latitude,
+                longitude, isTesting, request.contact!!.phone,
+                request.contact!!.email, request.contact!!.contactName
+            )
+        }
+
+        return when (id) {
+            null -> null
+            else -> IdWrapper(id)
+        }
     }
 
     @Path("/{id}/sites")
     @GET
     fun getSites(@PathParam("id") id: String): List<Any> {
-        TODO("not implemented")
+        return dao.getSites(id)!!
```

Comment:
if you are doing this, why would you define the method to return `null`? if you think it can't fail, change the signature to reflect that. if it can fail, don't return `null` as a failure signal, throw an exception